tests/test_salesforce_start_date.py

```diff
@@ -0,0 +1,147 @@
+                                      if catalog.get('tap_stream_id') in self.expected_sync_streams()]
+        self.perform_and_verify_table_and_field_selection(conn_id_2, test_catalogs_2_all_fields, select_all_fields=True)
+
+        # run sync
+        record_count_by_stream_2 = self.run_and_verify_sync(conn_id_2)
+
+        replicated_row_count_2 = sum(record_count_by_stream_2.values())
+        self.assertGreater(replicated_row_count_2, 0, msg="failed to replicate any data")
+        print("total replicated row count: {}".format(replicated_row_count_2))
+        synced_records_2 = runner.get_records_from_target_output()
+
+        for stream in self.expected_sync_streams():
+            with self.subTest(stream=stream):
+                replication_type = self.expected_replication_method().get(stream)
+
+                record_count_1 = record_count_by_stream_1.get(stream, 0)
+                record_count_2 = record_count_by_stream_2.get(stream, 0)
+
+                if replication_type == self.INCREMENTAL:
+                    replication_key = next(iter(self.expected_replication_keys().get(stream)))
+
+                    # Verify replication key is greater or equal to start_date for sync 1
+                    replication_dates_1 =[row.get('data').get(replication_key) for row in
+                                          synced_records_1.get(stream, {'messages': []}).get('messages', [])
+                                          if row.get('data')]
+                    for replication_date in replication_dates_1:
+                        self.assertGreaterEqual(
+                            self.parse_date(replication_date), self.parse_date(self.start_date_1),
+                                msg="Report pertains to a date prior to our start date.\n" +
+                                "Sync start_date: {}\n".format(self.start_date_1) +
+                                "Record date: {} ".format(replication_date)
+                        )
+
+                    # # Verify replication key is greater or equal to start_date for sync 2
+                    replication_dates_2 =[row.get('data').get(replication_key) for row in
+                                          synced_records_2.get(stream, {'messages': []}).get('messages', [])
+                                          if row.get('data')]
+                    for replication_date in replication_dates_2:
+                        self.assertGreaterEqual(
+                            self.parse_date(replication_date), self.parse_date(self.start_date_2),
+                                msg="Report pertains to a date prior to our start date.\n" +
+                                "Sync start_date: {}\n".format(self.start_date_2) +
+                                "Record date: {} ".format(replication_date)
+                        )
+
+
+                    # Verify that the 2nd sync with a later start date replicates the same number of
```

Comment:
This assertion and error message do not match.

Reply:
The assertion should be `assertLess` but this requires sufficient test data set prior to test running. Changed as `assertLessEqual` for now with a TODO pointed to a backlog ticket to fix the test data so we can run this assertion.